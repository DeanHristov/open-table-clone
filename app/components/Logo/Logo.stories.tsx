import * as React from "react";
import { Meta, Story } from "@storybook/react";
import Logo, { ILogoProps } from "./Logo";

export default {
  title: "Components/Logo",
  component: Logo,
} as Meta;

export const Default: Story<ILogoProps> = (args) => <Logo {...args} />;

Default.args = {
  width: 200,
  href: "/",
};
