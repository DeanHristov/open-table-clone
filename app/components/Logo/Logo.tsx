import { FC } from "react";
import Image from "next/image";
import Link from "next/link";

export interface ILogoProps {
  width?: number;
  href?: string;
}

const Logo: FC<ILogoProps> = ({ width = 200, href = "/" }) => {
  return (
    <div className="m-0 p-0">
      <Link href={href}>
        <Image
          src="/opentable-logo.svg"
          alt="Open table"
          width={width}
          height={width}
        />
      </Link>
    </div>
  );
};

export default Logo;
