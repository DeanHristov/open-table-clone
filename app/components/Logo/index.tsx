import Logo, { ILogoProps } from "./Logo";

export type { ILogoProps };
export default Logo;
