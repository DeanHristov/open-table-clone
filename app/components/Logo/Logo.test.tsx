import { render } from "@testing-library/react";
import Logo from "./Logo";

describe("Component: <Logo {...} />", () => {
  it("Should be able to render default state", () => {
    const { container } = render(<Logo />);

    expect(container).toMatchSnapshot();
  });
});
