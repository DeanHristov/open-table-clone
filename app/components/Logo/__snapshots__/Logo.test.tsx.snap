// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Logo {...} /> Should be able to render default state 1`] = `
<div>
  <div
    class="m-0 p-0"
  >
    <a
      href="/"
    >
      <img
        alt="Open table"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/opentable-logo.svg"
        style="color: transparent;"
        width="200"
      />
    </a>
  </div>
</div>
`;
